File: classify/two-pointers/remove-duplicates-from-sorted-array.c
int removeDuplicates(int* nums, int numsSize) {
    int count = 0;
    int cur = 0;

    while(cur < numsSize){
        if(cur == 0 || nums[cur] != nums[cur-1])
            nums[count++] = nums[cur];
        ++cur;
    }

    return count;
}

File: go/array/remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    i, j := 0, 0
    for j < len(nums) {
        for j < len(nums) && nums[i] == nums[j] {
            j++
        }
        if j < len(nums) {
            i++
            nums[i], nums[j] = nums[j], nums[i]
        }
        j++
    }

    return i + 1
}
